### DISCUSSION TITLE
Chaining .or filters

### DISCUSSION THREAD

**Discussion by rphly** [supabase/supabase#2324] [Questions]
I have a requirement to reduce information queried based on certain rules. 
Something like this:
```
supabase.from("table")
.or(predicate1, predicate2)
.or(predicate3, predicate4)
```
which I expect to be the rough equivalent of:
```
SELECT *
FROM table
WHERE
(predicate1 or predicate2)
AND (predicate3 or predicate4)
```

However, this is not available on `supabase-dart` as the params builder `appendSearchParams` will ensure only one .or statement is set. 

Is there a workaround for this that I'm missing? I require this for a realtime feature so I don't think a view or RPC is in the cards here.

**Comment by dshukertjr** [comment]
Hi @rphly!

So sorry for the late response. Currently, realtime only supports basic filtering in this format: {table}:{col}=eq.{val} - where {col} is the column name, and {val} is the value which you want to match. 

For example:
```dart
supabase
    .from('countries:id=eq.200')
    .on(SupabaseEventTypes.all, (payload) {})
    .subscribe();
```

`or` filter is part of postgrest library, so can not be used with realtime subscription as explained in the [filter documentation](https://supabase.io/docs/reference/javascript/using-filters). 

Maybe what you are trying to achieve can be done by filtering the data on the client side. Do you have any example of what each predicate could look like?

**Reply by rphly** [reply]
Ah got it. Yeah it's not very clear which filters are supported in real-time and not supported in real-time from the docs. 😅 "Basic" filtering could mean different things to different people. Thanks for the response!

Mmmm the predicates are really just lookign for specific values for specific columns. So we want the data returned to be exclusive for one particular column from another as demonstrated in my example SQL statement.

**Reply by dshukertjr** [reply]
Sorry for giving you the wrong impression, but "basic" is just a term that I used, and is not the official way of how Supabase describes the filtering capability of realtime. 

To achieve what you want to do, you can filter your data on the client side like this. 
```typescript
supabase
    .from('countries:id=eq.200')
    .on(SupabaseEventTypes.all, (payload) {
         // Check if the data is what you want
    })
    .subscribe();
```

**Reply by rphly** [reply]
No worries. Yes that is what I'm currently doing, until realtime supports RLS. Ideally I would have liked that the data is excluded before being sent over the network but this will do for now
